StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentMethodType.swift

```diff
@@ -115,21 +125,26 @@ extension PaymentSheet {
         ///   - configuration: A `PaymentSheet` configuration.
         static func filteredPaymentMethodTypes(from intent: Intent, configuration: Configuration, logAvailability: Bool = false) -> [PaymentMethodType]
         {
-            var recommendedStripePaymentMethodTypes = intent.recommendedPaymentMethodTypes
-
+            var additionalPaymentMethodTypes: [PaymentMethodType] = []
             if
-                recommendedStripePaymentMethodTypes.contains(.link),
-                !recommendedStripePaymentMethodTypes.contains(.USBankAccount),
+                intent.recommendedPaymentMethodTypes.contains(.link),
+                !intent.recommendedPaymentMethodTypes.contains(.USBankAccount),
                 !intent.isDeferredIntent,
                 intent.linkFundingSources?.contains(.bankAccount) == true
             {
-                // we must add this BEFORE we do filtering via
-                // `PaymentSheet.PaymentMethodType.supportsAdding`
-                // because we want to filter out instant debits
-                // IF the user has not linked Financial Connections
-                recommendedStripePaymentMethodTypes.append(.instantDebits)
+                let availabilityStatus = configurationSatisfiesRequirements(
+                    requirements: [.financialConnectionsSDK],
```

Comment:
Some qs:
- OOC, why would a merchant have the financialConnectionsSDK enabled but not have USBankAccount?
- is instantDebits (or really, link + US Bank Account) a delayed payment method? i.e. is the Intent status `succeeded` immediately after confirming the Intent and handling next actions, or does it take time to process?

Reply:
> OOC, why would a merchant have the financialConnectionsSDK enabled but not have USBankAccount?

I'm likely not understanding the question 100%, _but_ the first order of business in terms of checking for `financialConnectionsSDK` is just to do a client-side check whether merchant has linked Financial Connections SDK. Because instant debits depends FC SDK, this, in a very simple manner, makes sense to check. From a very first-principles manner, its unrelated to anything else.

> is instantDebits (or really, link + US Bank Account) a delayed payment method? i.e. is the Intent status succeeded immediately after confirming the Intent and handling next actions, or does it take time to process?

If I understood the question correctly, Instant Debits is a special payment method that allows for instant confirmation. So its US Bank/ACH, except "instant" (for the merchant - Stripe handles operation burden of risk/returns [doc](https://docs.google.com/document/d/19HSmr7KV-JjDaq28kwDOi1CRQwsZGY7nc0H1ph7Gbss/edit#heading=h.1wve2hxdcps6)).